modules/core/src/main/java/org/apache/ignite/internal/processors/metastorage/persistence/DmsDataWriterWorker.java

```diff
@@ -81,134 +90,125 @@ public DmsDataWriterWorker(
         super(igniteInstanceName, "dms-writer", log);
         this.lock = lock;
         this.errorHnd = errorHnd;
+
+        pauseTask = new FutureTask<>(() -> AWAIT);
+        // Completed future task.
+        pauseTask.run();
+
+        // Put restore task to the queue, so it will be executed on worker start.
+        updateQueue.offer(newDmsTask(this::restore));
     }
 
     /** */
     public void setMetaStorage(ReadWriteMetastorage metastorage) {
         this.metastorage = metastorage;
     }
 
-    /** */
-    public void update(DistributedMetaStorageHistoryItem histItem) {
-        updateQueue.offer(histItem);
+    /**
+     * @return Future which will be completed will all the tasks prior to the pause task completed.
```

Comment:
Future which will be completed ```diff - will``` all the tasks prior to the pause task completed.

Reply:
Fixed.